pkg/declarativeconfig/configuration.go

```diff
@@ -111,3 +100,37 @@ func decodeYAMLToConfiguration(rawYAML []byte, configuration Configuration) erro
 	}
 	return nil
 }
+
+func parseToConfiguration(contents []byte) ([]Configuration, error) {
+	dec := yaml.NewDecoder(bytes.NewReader(contents))
+	var unstructuredObjs []interface{}
+	for {
+		var obj interface{}
+		if dec.Decode(&obj) != nil {
```

Comment:
This code ignores unexpected errors. How about we `break` without an error only on `io.EOF` returned?